src/journal/src/Journal/Types.hs

```diff
@@ -191,6 +197,27 @@ readPageSize (Metadata meta) = readInt32OffAddr meta lOG_PAGE_SIZE_OFFSET
 writePageSize :: Metadata -> Int32 -> IO ()
 writePageSize (Metadata meta) = writeInt32OffAddr meta lOG_PAGE_SIZE_OFFSET
 
+readBytesConsumed :: Metadata -> IO Int
```

Comment:
I wonder if it perhaps makes more sense to use a fixed size int here?

Reply:
Maybe, but that would happen in another PR.